recipes/openmpi/all/conanfile.py

```diff
@@ -0,0 +1,70 @@
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class OpenMPIConan(ConanFile):
+    name = "openmpi"
+    homepage = "https://www.open-mpi.org"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("conan", "mpi", "openmpi")
+    description = "A High Performance Message Passing Library"
+    license = "https://www.open-mpi.org/community/license.php"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "fortran": ['yes', 'mpifh', 'usempi', 'usempi80', 'no']}
+    default_options = {'shared': False, 'fPIC': True, 'fortran': 'no'}
+    _source_subfolder = "sources"
+
+    def config(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("OpenMPI doesn't support Windows")
+
+    def requirements(self):
+        self.requires("zlib/1.2.11")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def build(self):
+        with tools.chdir(self._source_subfolder):
+            env_build = AutoToolsBuildEnvironment(self)
+            env_build.fpic = self.options.fPIC
+            args = ['--disable-wrapper-rpath', '--disable-wrapper-runpath']
+            if self.settings.build_type == 'Debug':
+                args.append('--enable-debug')
+            if self.options.shared:
+                args.extend(['--enable-shared', '--disable-static'])
+            else:
+                args.extend(['--enable-static', '--disable-shared'])
+            args.append('--with-pic' if self.options.fPIC else '--without-pic')
+            args.append('--enable-mpi-fortran=%s' % str(self.options.fortran))
+            args.append('--with-zlib=%s' % self.deps_cpp_info['zlib'].rootpath)
+            args.append('--with-zlib-libdir=%s' % self.deps_cpp_info['zlib'].lib_paths[0])
+            args.append('--datarootdir=${prefix}/res')
+            env_build.configure(args=args)
+            env_build.make()
+            env_build.install()
+
+    def package(self):
+        self.copy(pattern="LICENSE", src='sources', dst='licenses')
+
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "etc"))
+        tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la")
+
+    def package_info(self):
```

Comment:
Is it possible to properly model this module file: https://cmake.org/cmake/help/latest/module/FindMPI.html?

It is called in test_package.

Reply:
I didn't try, that's too much for the first iteration, as Find module is pretty complex - mpi packages not just libraries, but compiler wrappers

Reply:
Therefore could you add a FIXME for CMake and pkg-config?

```
/usr/bin/install -c -m 644 ompi.pc ompi-c.pc ompi-cxx.pc ompi-fort.pc '/home/conan/w/cci_PR-4482/.conan/data/openmpi/4.1.0/_/_/package/b2c1c978ecb1a3aa8058e2e05f1dcb8093dfc54a/lib/pkgconfig'
```

Reply:
what I have right now:
```
[HOOK - conan-center.py] post_package(): ERROR: [PC-FILES (KB-H020)] Found files:
./lib/pkgconfig/ompi-cxx.pc
./lib/pkgconfig/orte.pc
./lib/pkgconfig/ompi-c.pc
./lib/pkgconfig/ompi-f90.pc
./lib/pkgconfig/ompi-f77.pc
./lib/pkgconfig/pmix.pc
./lib/pkgconfig/ompi-fort.pc
./lib/pkgconfig/ompi.pc
```
maybe it's easy to mimic in components, I don't have much experience with it?

Reply:
I guess yes. The content of each .pc file would help to see what is the associated lib, and inter dependencies.